### PR TITLE
Register the services as Spring beans with reasonable name

### LINKED ISSUE #458: rename service beans registered automatically
Not to have names like `getProjectTemplateService` but only `projectTemplateService`.

### PR DESCRIPTION
**Pull request by jimirocks**
Resolves #458

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: gooddata-java/src/main/java/com/gooddata/sdk/service/GoodData.java [hunk 1]
```diff
@@ -335,7 +335,7 @@ public void logout() {
      *
      * @return initialized service for project management
      */
-    @Bean
+    @Bean("goodDataprojectService")
```
Comment (liry) [inline gooddata-java/src/main/java/com/gooddata/sdk/service/GoodData.java]: "P" should be upper case

File: gooddata-java/src/test/groovy/com/gooddata/sdk/service/GoodDataBeansAsServicesIT.groovy [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import spock.lang.Shared
+import spock.lang.Specification
+import spock.lang.Unroll
+
+class GoodDataBeansAsServicesIT extends Specification {
+
+    @Shared ClassPathXmlApplicationContext context = new ClassPathXmlApplicationContext('spring/gd-annotationConfig.xml')
+
+    @Unroll
+    def "should register service of type #clazzName from xml using annotation config"() {
```
Comment (liry) [inline gooddata-java/src/test/groovy/com/gooddata/sdk/service/GoodDataBeansAsServicesIT.groovy]: from xml using annotation config?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: gooddata-java/src/main/java/com/gooddata/sdk/service/GoodData.java (hunk 1 above)
Comment (jimirocks) [reply gooddata-java/src/main/java/com/gooddata/sdk/service/GoodData.java]: fixed

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-java-merge-compliance-check/60/ : SUCCESS in 27s

**Pull request merged**